FerretDB/FerretDB
# Add `saslStart` stub

# Description

Closes #1593.

We cannot enable authorization to run both FerretDB and MongoDB with `PLAIN` authorization mechanism because it is not available for MongoDB. Implemented stub was tested locally with FerretDB. 
To test it locally you need:
* checkout into this branch;
* execute `bin/task run`
* execute in another terminal `docker compose exec mongodb mongosh 'mongodb://username:password@host.docker.internal:27018/test?tls=true&tlsCAFile=/etc/certs/rootCA.pem&heartbeatFrequencyMS=300000&authMechanism=PLAIN'`

In the FerretDB output check for the `saslStart` string. FerretDB will respond with `ok`, MongoDB will respond with `MechanismUnavailable` error.



## Readiness checklist



* [ ] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.
